### PR TITLE
feat: [modal migration] upvoted by

### PR DESCRIPTION
**Pull request by el**
## Changes

Reopened PR for upvoted by

### CODE REVIEW

**Review by rebelchris — CHANGES_REQUESTED** [review]
Specific issues with this modal:

- Size seems to wide at the moment
- Backdrop (as handled in other task, but please merge that one before or pull that change in?)
- The user items have padding
- The user items don't show the user tooltip on hover

Old:
Screenshot 2022-11-17 at 10 03 22

New:
Screenshot 2022-11-17 at 10 03 29

File: packages/shared/src/components/CloseButton.tsx [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+  Button,
+  StyledButtonProps,
+  ButtonProps,
+  ButtonSize,
+} from './buttons/Button';
 import CloseIcon from './icons/Close';
 
+type CloseButtonProps = StyledButtonProps &
+  ButtonProps<'button'> & {
+    size?: ButtonSize;
```
Comment (sshanzel) [inline packages/shared/src/components/CloseButton.tsx:13]: Should we not keep using the `buttonSize` as it is already available from the `ButtonProps`

File: packages/shared/src/components/modals/common/ModalClose.tsx [hunk 2]
```diff
@@ -13,7 +13,8 @@ function ModalCloseComponent(
       {...props}
       onClick={onClick}
       ref={ref}
-      className={classNames('right-4 z-1', className)}
+      size="medium"
```
Comment (sshanzel) [inline packages/shared/src/components/modals/common/ModalClose.tsx:16]: I think we can just use the `buttonSize` instead of introducing a new one.

**Review by sshanzel — COMMENTED** [review]
Code-wise I only had one comment (around button size property), other than that. There is nothing more I can add. Chris has already mentioned the concerns. 

Thank you for taking this on. Implementations of modal looks so clean 🚢

File: packages/shared/src/components/CloseButton.tsx (hunk 1 above)
Comment (el) [reply packages/shared/src/components/CloseButton.tsx:13]: Great catch. Fixed 👍

File: packages/shared/src/components/modals/common/ModalClose.tsx (hunk 2 above)
Comment (el) [reply packages/shared/src/components/modals/common/ModalClose.tsx:16]: Yeap, sharp eyes 👀

**Review by rebelchris — APPROVED** [review]
Tested and looks to work perfectly now 🥳

**Pull request merged**